internal/gengapic/gengrpc.go

```diff
@@ -177,7 +177,9 @@ func (g *generator) grpcClientOptions(serv *descriptor.ServiceDescriptorProto, s
 	p("func default%sGRPCClientOptions() []option.ClientOption {", servName)
 	p("  return []option.ClientOption{")
 	p("    internaloption.WithDefaultEndpoint(%q),", host)
+	p("    internaloption.WithDefaultEndpointTemplate(%q),", generateDefaultEndpointTemplate(host))
 	p("    internaloption.WithDefaultMTLSEndpoint(%q),", generateDefaultMTLSEndpoint(host))
+	p("    internaloption.WithDefaultUniverseDomain(%q),", googleDefaultUniverse)
```

Comment:
Without knowing much how this will ultimately work, do the `WithDefaultEndpoint` and `WithDefaultMTLSEndpoint` options go away assuming endpoint resolution is done by the middleware?

Reply:
`WithDefaultEndpoint` is already deprecated, and will eventually go away post-transition. `WithDefaultMTLSEndpoint` is unaffected by these changes as that auth flow does not support universe domain.